vowpalwabbit/array_parameters.h

```diff
@@ -0,0 +1,181 @@
+#pragma once
+#include <stdio.h>
+#include <iostream>
+#include <stdlib.h>
+#include <string.h>
+
+#ifndef _WIN32
+#include <sys/mman.h>
+#endif
+
+using namespace std;
+typedef float weight;
+
+class weight_vector;
+
+class weights_iterator_iterator
```

Comment:
I used the class when there is an outer for loop that moves with a stride and the inner loop moves within the bucket -
```
for (uint64_t j = 0; j < stride * length; j += stride)
 for (size_t k = 0; k < all->lda; k++)
   all->reg.weight_vector[j + k] = (float)(-log(frand48()) + 1.0f);  
```
(Line 634-642 in lda_core.cc)